src/components/Image/index.js

```diff
@@ -7,67 +7,58 @@ import ONYXKEYS from '../../ONYXKEYS';
 import {defaultProps, imagePropTypes} from './imagePropTypes';
 import RESIZE_MODES from './resizeModes';
 
-class Image extends React.Component {
-    componentDidMount() {
-        this.configureOnLoad();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.source === this.props.source) {
-            return;
-        }
-        this.configureOnLoad();
-    }
-
+function Image(props) {
     /**
      * Check if the image source is a URL - if so the `encryptedAuthToken` is appended
      * to the source.
      * @returns {Object} - the configured image source
      */
-    getImageSource() {
-        const source = this.props.source;
+    const getImageSource = useCallback(() => {
+        const source = props.source;
         let imageSource = source;
-        if (this.props.isAuthTokenRequired) {
+        if (props.isAuthTokenRequired) {
             // There is currently a `react-native-web` bug preventing the authToken being passed
             // in the headers of the image request so the authToken is added as a query param.
             // On native the authToken IS passed in the image request headers
-            const authToken = lodashGet(this.props, 'session.encryptedAuthToken', null);
+            const authToken = lodashGet(props, 'session.encryptedAuthToken', null);
             imageSource = {uri: `${source.uri}?encryptedAuthToken=${encodeURIComponent(authToken)}`};
         }
 
         return imageSource;
-    }
+    }, [props]);
 
     /**
      * The natural image dimensions are retrieved using the updated source
      * and as a result the `onLoad` event needs to be manually invoked to return these dimensions
      */
-    configureOnLoad() {
+    const configureOnLoad = useCallback(() => {
         // If an onLoad callback was specified then manually call it and pass
         // the natural image dimensions to match the native API
-        if (this.props.onLoad == null) {
+        if (props.onLoad == null) {
             return;
         }
 
-        const imageSource = this.getImageSource();
+        const imageSource = getImageSource();
         RNImage.getSize(imageSource.uri, (width, height) => {
-            this.props.onLoad({nativeEvent: {width, height}});
+            props.onLoad({nativeEvent: {width, height}});
         });
-    }
+    }, [getImageSource, props]);
+
+    useEffect(() => {
+        configureOnLoad();
+    }, [configureOnLoad, props]);
 
-    render() {
-        // Omit the props which the underlying RNImage won't use
-        const forwardedProps = _.omit(this.props, ['source', 'onLoad', 'session', 'isAuthTokenRequired']);
-        const source = this.getImageSource();
+    // Omit the props which the underlying RNImage won't use
+    const forwardedProps = _.omit(props, ['source', 'onLoad', 'session', 'isAuthTokenRequired']);
+    const source = getImageSource();
```

Comment:
when you will switch getImageSource to useMemo - you can directly add it on line 59 without any new variable